### PR TITLE
Offer Api Key validation in SDK

### PR DESCRIPTION
**Pull request by rain-on**
Octopus CI plugins are all required to validate API Keys - to ensure a consistent response to invalid keys, the validation of said key has been moved into the SDK, such that it can be reused in all plugins.

### CODE REVIEW

File: octopus-sdk/src/test/java/com/octopus/sdk/utils/ApiKeyValidatorTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  public void keyContainingNoAlphaNumericCharactersfails() {
+    assertThatThrownBy(() -> ApiKeyValidator.validate("API-1234567890ABCDEFGHIJKLM-0"))
+        .isInstanceOf(IllegalArgumentException.class);
+  }
+
+  @Test
+  public void validKeyDoesNotThrow() {
+    ApiKeyValidator.validate("API-1234567890ABCDEFGHIJKLMNO");
+  }
+}
```
Comment (long-sean) [inline octopus-sdk/src/test/java/com/octopus/sdk/utils/ApiKeyValidatorTest.java:52]: Test the negative case (invalid character/s)

File: octopus-sdk/src/main/java/com/octopus/sdk/utils/ApiKeyValidator.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+package com.octopus.sdk.utils;
+
+public class ApiKeyValidator {
+
+  public static void validate(String apiKeyValue) {
+    final String API_KEY_PREFIX = "API-";
+    if (!apiKeyValue.startsWith(API_KEY_PREFIX)) {
+      throw new IllegalArgumentException("APi Key must start with " + API_KEY_PREFIX);
```
Comment (long-sean) [inline octopus-sdk/src/main/java/com/octopus/sdk/utils/ApiKeyValidator.java:23]: Capitalisation in "APi..."

File: octopus-sdk/src/test/java/com/octopus/sdk/utils/ApiKeyValidatorTest.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.octopus.sdk.utils;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import org.junit.jupiter.api.Test;
+
+class ApiKeyValidatorTest {
```
Comment (long-sean) [inline octopus-sdk/src/test/java/com/octopus/sdk/utils/ApiKeyValidatorTest.java:22]: This is another instance where I think asserting exception message text is important.
1. Help capture typos / formatting
2. The exception text is the only thing that makes assertions in this test class unique

Happy to discuss

**Review by long-sean — APPROVED** [review]

File: octopus-sdk/src/main/java/com/octopus/sdk/utils/ApiKeyValidator.java (hunk 2 above)
Comment (rain-on) [reply octopus-sdk/src/main/java/com/octopus/sdk/utils/ApiKeyValidator.java]: done

**Pull request merged**